features/netsocket/TCPSocket.cpp

```diff
@@ -19,8 +19,7 @@
 #include "mbed_assert.h"
 
 TCPSocket::TCPSocket()
-    : _pending(0), _read_sem(0), _write_sem(0),
-      _read_in_progress(false), _write_in_progress(false)
+    : _pending(0), _read_sem(0), _write_sem(0)
```

Comment:
Come on @geky! I told you these are needed for TCP. If you want this functionality then add it the right way by adding the functions send_all and receive_all. Don't encourage code which can easily lead to bugs.